### PR TITLE
Tests: Add test for getdifficulty

### PR DESCRIPTION
**Pull request by jimmysong**
Test added to blockchain.py as adding a new test to reduce test run time.

### CODE REVIEW

File: test/functional/blockchain.py [hunk 1]
```diff
@@ -57,7 +64,8 @@ def _test_gettxoutsetinfo(self):
     def _test_getblockheader(self):
         node = self.nodes[0]
 
-        assert_raises_jsonrpc(-5, "Block not found", node.getblockheader, "nonsense")
+        assert_raises_jsonrpc(-5, "Block not found",
```
Comment (paveljanik) [inline test/functional/blockchain.py]: Why this change?

File: test/functional/blockchain.py [hunk 2]
```diff
@@ -40,8 +40,15 @@ def setup_network(self, split=False):
     def run_test(self):
         self._test_gettxoutsetinfo()
         self._test_getblockheader()
+        self._test_getdifficulty()
         self.nodes[0].verifychain(4, 0)
 
+    def _test_getdifficulty(self):
+        res = self.nodes[0].getdifficulty()
```
Comment (paveljanik) [inline test/functional/blockchain.py]: Mention the `getdifficulty` RPC test in the top level comment or remove the list of tested methods there.

File: test/functional/blockchain.py [hunk 3]
```diff
@@ -40,8 +40,15 @@ def setup_network(self, split=False):
     def run_test(self):
         self._test_gettxoutsetinfo()
         self._test_getblockheader()
+        self._test_getdifficulty()
         self.nodes[0].verifychain(4, 0)
 
+    def _test_getdifficulty(self):
```
Comment (paveljanik) [inline test/functional/blockchain.py]: Can you move the def after the `_test_getblockheader` so it is ordered the same as it is called?

File: test/functional/blockchain.py [hunk 4]
```diff
@@ -40,8 +40,15 @@ def setup_network(self, split=False):
...
+        res = self.nodes[0].getdifficulty()
+        # 1 hash in 2 should be valid, so difficulty should be 1/2**31
+        # binary => decimal => binary math is why we round
+        assert_equal(round(res * 2**31, 4), 1.0)
```
Comment (paveljanik) [inline test/functional/blockchain.py]: Ugh.

**Review by paveljanik — CHANGES_REQUESTED** [review]

File: test/functional/blockchain.py (hunk 3 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: Yep, will do.

File: test/functional/blockchain.py (hunk 2 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: adding to the comment, thanks!

File: test/functional/blockchain.py (hunk 1 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: flake8 warning. I can change this back?

File: test/functional/blockchain.py (hunk 4 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: Is there a cleverer way to do this?

**Comment by jimmysong** [comment]
Fixed issues, rebased to master.

**Review by paveljanik — APPROVED** [review]
ACK 821dd5e

File: test/functional/blockchain.py [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 Test the following RPCs:
     - gettxoutsetinfo
+    - getdifficulty
     - verifychain
```
Comment (jnewbery) [inline test/functional/blockchain.py]: Might as well fix this docstring while you're at it. The following RPCs are also tested:

- getbestblockhash
- getblockhash
- getblockheader

File: test/functional/blockchain.py (hunk 1 above)
Comment (jnewbery) [inline test/functional/blockchain.py]: My personal preference is to ignore the 'line too long' warning. Nothing wrong with going over 79 character lines in my opinion (but we don't have a style guide for functional tests, so entirely up to you)

File: test/functional/blockchain.py [hunk 6]
```diff
@@ -79,5 +82,11 @@ def _test_getblockheader(self):
         assert isinstance(int(header['versionHex'], 16), int)
         assert isinstance(header['difficulty'], Decimal)
 
+    def _test_getdifficulty(self):
+        res = self.nodes[0].getdifficulty()
```
Comment (jnewbery) [inline test/functional/blockchain.py]: I'd prefer a more descriptive name than `res`, perhaps `difficulty`? (or even combine with the line below)

**Review by jnewbery — COMMENTED** [review]
Looks good. Tested ACK ba2b295093d5f5cad4701f9ca86b2c83e60c3ef9.

A few nits which you can take or leave.

**Review by jnewbery — APPROVED** [review]

File: test/functional/blockchain.py (hunk 5 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: good idea.

File: test/functional/blockchain.py (hunk 6 above)
Comment (jimmysong) [inline test/functional/blockchain.py]: Makes sense.

**Comment by jimmysong** [comment]
Fixed nits and squashed.
